src/tracks/playabletrack/wavetrack/ui/WaveTrackView.cpp

```diff
@@ -1252,6 +1255,17 @@ void WaveTrackView::Reparent( const std::shared_ptr<Track> &parent )
    WaveTrackSubViews::ForEach( [&parent](WaveTrackSubView &subView){
       subView.Reparent( parent );
    } );
+   if (mpAffordanceCellControl)
+      mpAffordanceCellControl->Reparent(parent);
+}
+
+std::weak_ptr<WaveClip> WaveTrackView::GetSelectedClip()
+{
+   if (auto affordance = std::dynamic_pointer_cast<WaveTrackAffordanceControls>(GetAffordanceControls()))
```

Comment:
I think that could safely be a static cast.  WaveTrackView controls the definition of its affordance cell as always of this type.

Reply:
Still unresolved